presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -825,6 +825,19 @@ public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Expression> cont
             postJoinConjuncts.addAll(equalityPartition.getScopeComplementEqualities());
             postJoinConjuncts.addAll(equalityPartition.getScopeStraddlingEqualities());
 
+            if (extractConjuncts(inheritedPredicate).contains(node.getSemiJoinOutput().toSymbolReference())) {
```

Comment:
you need to check in what context `semijoin` symbol is used, e.g:
`semijoin and y`
`not semijoin`
`semijoin = func_that_returns_boolean(...)`